controller/appcontroller.go

```diff
@@ -171,43 +226,29 @@ func toString(val interface{}) string {
 // hideSecretData checks if given object kind is Secret, replaces data keys with stars and returns unchanged data map. The method additionally check if data key if different
 // from corresponding key of optional parameter `otherData` and adds extra star to keep information about difference. So if secret data is out of sync user still can see which
 // fields are different.
-func hideSecretData(state string, otherData map[string]interface{}) (string, map[string]interface{}) {
-	obj, err := appv1.UnmarshalToUnstructured(state)
-	if err == nil {
-		if obj != nil && obj.GetKind() == kube.SecretKind {
-			if data, ok, err := unstructured.NestedMap(obj.Object, "data"); err == nil && ok {
-				unchangedData := make(map[string]interface{})
-				for k, v := range data {
-					unchangedData[k] = v
-				}
-				for k := range data {
-					replacement := "********"
-					if otherData != nil {
-						if val, ok := otherData[k]; ok && toString(val) != toString(data[k]) {
-							replacement = replacement + "*"
-						}
+func hideSecretData(state *unstructured.Unstructured, otherData map[string]interface{}) (*unstructured.Unstructured, map[string]interface{}) {
+	obj := state.DeepCopy()
+	if obj != nil && obj.GetKind() == kube.SecretKind {
```

Comment:
To be safe, I think this we should also check for group, in case theres a CRD with Kind Secret.

Reply:
Good catch! Done: https://github.com/argoproj/argo-cd/pull/840/commits/a842eb3bb31f378c37866ea5deba3efc150d4341